book/01-javascript-necessary-knowledge-base.md

```diff
@@ -654,3 +658,229 @@ console.log(array.some(even));
 ```
 
 Mình xin giới thiệu đến đây thôi, còn những function khác như `every()`, `find()`, `includes()`,... các bạn tìm hiểu thêm ở [đây](https://developer.mozilla.org/vi/docs/Web/JavaScript/Reference/Global_Objects/Array) nha.
+
+## 9. Callback, Promise, Async and Await
+
+### 9.1. Callback
+
+Một callback function là function được gọi bởi function khác. Thông thường, function A (callback function) sẽ được truyền vào function B như một parameter,
+function B sẽ gọi function A để thực hiện một chức năng nào đó.
+
+```javascript
+function A() {
+  // ...........
+}
+
+function B(callbackFn) {
+  // ..........
+  callbackFn();
+}
+
+B(A);
+```
+
+Được sử dụng để xử lý bất đồng bộ hoặc xử lý sự kiện
+
+```javascript
+$("#test").click(() => {
```

Comment:
do not give an example by using jquery.
May use these following suggestions:
- normal js function
- setTimeout()
- addEventListener()
- react Hook
Give an example by using arrow function and regular function